cosmoz-default-tree.html

```diff
@@ -23,134 +32,125 @@
+		var node = this.getNodeByProperty(propertyName, propertyValue, exact, all, nodeList);
+		if (!node) {
 			return;
 		}
-		return this._getPathByPathLocator(node.pathLocator);
+		return this.getPathString(node.path, propertyName);
+	},
+
+	/**
+	 * Loops over a node list (or roots if not defined) and returns the concatenated results of search().
+	 */
+	Cosmoz.DefaultTree.prototype._searchNodeList = function (propertyName, propertyValue, exact, all, nodeList) {
+		var results = [];
+
+		// Defaults
+		exact = exact !== undefined ? exact : true;
+		all = all !== undefined ? all : false;
+		nodeList = nodeList ? nodeList : this._roots;
+
+		nodeList.forEach(function (node) {
+			results = results.concat(this.search(node, propertyName, propertyValue, exact, all));
+		}, this);
+			
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue) {
-		if (propertyName === undefined || propertyValue === undefined) {
-			return;
+	/**
+	 * Returns the Node of a given pathLocator.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
+		var	nodesOnPath = this.getPath(pathLocator, this._treeData);
+		return nodesOnPath[nodesOnPath.length - 1];
+	};
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+	/**
+	 * Returns an Array of nodes on a given path.
+	 * If pathLocator is empty or not defined, the Root objects get returned.
+	 */
+	Cosmoz.DefaultTree.prototype.getPath = function (pathLocator, nodes, separatorSign) {
+		if (!pathLocator) {
+			return this._roots
+		};
+
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.'
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodes ? nodes : this._treeData;
+		
+		return path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = this.getChildren(node);
+			node['key'] = nodeKey;
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
+	},
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+	/**
+	 * Returns a String (separated node comparison properties) representing the path.
+	 */
+	Cosmoz.DefaultTree.prototype.getPathString = function (pathLocator, comparisonProperty, pathSeparator) {
+		var nodesOnPath = this.getPath(pathLocator, this._treeData),
+			path;
+		
+		// Defaults
+		pathSeparator = pathSeparator ? pathSeparator : '/';
+
+		return nodesOnPath.map(function (node) {
+			return node[comparisonProperty];
+		}).join(pathSeparator);
 	};
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
+	/**
+	 * Returns an Object or an Array representing the children of a node.
```

Comment:
Again, multiple return types.
Any code doing `this.getChildren(node)` should be able to do `node.children` instead so drop the `asList` argument and consider it always true (revert to old return statement)